spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,177 @@
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Andrew Rouse
+
+== Integration with Microprofile Metrics
+
+When Microprofile Fault Tolerance and Microprofile Metrics are used together, metrics are automatically added for each of
+the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker` or `@Bulkhead` annotation.
+
+=== Names
+
+The automatically added metrics follow a consistent pattern which includes the fully qualified name of the annotated method.
+In the tables below, the placeholder `<name>` should be replaced by the fully qualified method name with all `.` characters replaced with `_`.
+
+=== Metrics added for `@Retry`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_retry_calls_success_no_retry_total`
+| Counter | None
+| The number of times the method was called and succeeded without retrying
+
+|`ft_<name>_retry_calls_success_retry_total`
+| Counter | None
+| The number of times the method was called and succeeded after retrying at least once
+
+|`ft_<name>_retry_calls_failure_total`
+| Counter | None
+| The number of times the method was called and ultimately failed after retrying
+
+|`ft_<name>_retry_retries_total`
+| Counter | None
+| The total number of times the method was retried
+|===
+
+=== Metrics added for `@Timeout`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_timeout_execution_duration`
+| Histogram | Nanoseconds
+| Histogram of execution times for the method
+
+|`ft_<name>_timeout_calls_timed_out_total`
+| Counter | None
+| The number of times the method timed out
+
+|`ft_<name>_timeout_calls_not_timed_out_total`
+| Counter | None
+| The number of times the method completed without timing out
+|===
+
+=== Metrics added for `@CircuitBreaker`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_circuitbreaker_calls_succeeded_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that returned successfully
+
+|`ft_<name>_circuitbreaker_calls_failed_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that then failed
+
+|`ft_<name>_circuitbreaker_calls_prevented_total`
+| Counter | None
+| Number of calls prevented from running by an open circuit breaker
+
+|`ft_<name>_circuitbreaker_accumulated_open`
```

Comment:
I suggest to rename to ft_<name>_circuitbreaker_open_time_total so that "total" is at the end and "open" is closer to the beginning. Also, I think that "time total" is more descriptive than "accumulated"

Apply same for half open and closed

Reply:
`ft_<name>_circuitbreaker_time_open` prometheus output will append `_seconds`